### PR TITLE
feat(atomic): loading animation for segmented facets

### PR DESCRIPTION
**Pull request by mbenkirane-coveo**
https://coveord.atlassian.net/browse/KIT-185

https://user-images.githubusercontent.com/97299176/178804722-5a88670f-d398-494e-a366-abc23ed260f6.mov

### CODE REVIEW

**Comment by ThibodeauJF** [comment]
I thought the placeholder would be on the facet and not the scrollable element

**Comment by mbenkirane-coveo** [comment]
> I thought the placeholder would be on the facet and not the scrollable element

Would result in pretty much the same thing. I could change it.

**Comment by ThibodeauJF** [comment]
Yeah this way there will still be placeholders when devs use only the facet without the scrollable container

**Comment by mbenkirane-coveo** [comment]
> Yeah this way there will still be placeholders when devs use only the facet without the scrollable container

Sure !

File: packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx [hunk 1]
```diff
@@ -107,20 +107,36 @@ export class AtomicSegmentedFacetScrollable implements InitializableComponent {
 
   private renderArrowClass(direction: ArrowDirection) {
     if (direction === 'left') {
-      return 'left-0 ' + (this.hideLeftArrow ? 'invisible opacity-0' : '');
+      return (
+        'left-0 ' +
+        (this.hideLeftArrow || !this.searchStatusState.firstSearchExecuted
```
Comment (mbenkirane-coveo) [inline packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx:112]: I had to check for `!this.searchStatusState.firstSearchExecuted` to make sure the arrows don't load before the segmented facets.

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx [hunk 2]
```diff
@@ -148,16 +148,20 @@ export class AtomicSegmentedFacet
...
-    ) {
-      return <Hidden></Hidden>;
-    }
-
+  renderLoadingAnimation() {
     return (
+      <div
+        part="placeholder"
+        aria-hidden
+        class="h-8 my-2 w-screen bg-neutral animate-pulse"
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx:156]: would be nice if the width is related to the number of values of the facet

File: packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx:112]: could you put this in the render method instead for example

```typescript
render() {
    if (this.searchStatus.state.hasError) {
      return <Hidden></Hidden>;
    }
    return (
      <div part="scrollable-container" class="flex relative">
        {this.searchStatusState.firstSearchExecuted && this.renderArrow('left')}
        <div
          part="horizontal-scroll"
          class="wrapper-segmented flex flex-row overflow-x-scroll scroll-smooth"
          ref={(el) => (this.horizontalScroll = el as HTMLDivElement)}
        >
          <slot></slot>
        </div>
        {this.searchStatusState.firstSearchExecuted && this.renderArrow('right')}
      </div>
    );
}
```

File: packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx [hunk 3]
```diff
@@ -148,16 +148,20 @@ export class AtomicSegmentedFacet
...
+      ></div>
+    );
+  }
+
+  renderContainer() {
+    return !this.facetState.values.length ? (
+      <Hidden></Hidden>
+    ) : (
```
[the remark this answers is not in the document]
Comment (mbenkirane-coveo) [reply packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx:164]: For some reason, if I don't put the state prop in a ternary, the returned result won't be updated unless the component rerenders.

File: packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx (hunk 2 above)
Comment (mbenkirane-coveo) [reply packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx:156]: I can do that !

Comment (mbenkirane-coveo) [reply packages/atomic/src/components/search/facets/atomic-segmented-facet/atomic-segmented-facet.tsx:156]: This is a bit more complex than what I thought. I will create a Jira for it.

**Pull request merged**